sdlib/dmd/gc.d

```diff
@@ -110,3 +140,59 @@ BlkInfo __sd_gc_druntime_allocInfo(void* ptr) {
 
 	return result;
 }
+
+void __sd_getArrayMetadata(void* ptr, void** base, size_t* size,
+                           size_t* flags) {
+	// all blocks are appendable in SDC GC.
+	auto pd = threadCache.maybeGetPageDescriptor(ptr);
+	auto e = pd.extent;
+	if (e) {
+		if (pd.isSlab()) {
+			auto si = SlabAllocInfo(pd, ptr);
+			*base = cast(void*) si.address;
+			*size = si.slotCapacity;
+			*flags = pd.toLeafPayload();
+		} else {
+			*base = e.address;
+			*size = e.size;
+			*flags = pd.toLeafPayload();
+		}
+	}
+}
+
+bool __sd_setArrayUsed(void* ptr, size_t pdData, size_t newUsed,
+                       size_t existingUsed, bool atomic) {
+	auto pd = PageDescriptor(pdData);
+	auto e = pd.extent;
+	if (e) {
+		if (pd.isSlab()) {
+			auto si = SlabAllocInfo(pd, ptr);
+			if (existingUsed == -1 || si.usedCapacity == existingUsed) {
+				si.setUsedCapacity(newUsed);
+				return true;
+			}
+		} else {
+			if (existingUsed == -1 || e.usedCapacity == existingUsed) {
+				e.setUsedCapacity(newUsed);
+				return true;
+			}
+		}
+	}
+
+	return false;
+}
+
+size_t __sd_getArrayUsed(void* ptr, size_t pdData, bool atomic) {
+	auto pd = PageDescriptor(pdData);
+	auto e = pd.extent;
+	if (e) {
+		if (pd.isSlab()) {
+			auto si = SlabAllocInfo(pd, ptr);
+			return si.usedCapacity;
+		} else {
+			return e.usedCapacity;
+		}
+	}
+
+	return 0;
+}
```

Comment:
This does not look like it is necessary to destroy anything.

Reply:
It's for the array runtime stuff, which can be a separate PR. Removed.